includes/class-amp-theme-support.php

```diff
@@ -14,6 +14,83 @@ class AMP_Theme_Support {
 
 	const COMPONENT_SCRIPTS_PLACEHOLDER = '<!--AMP_COMPONENT_SCRIPTS_PLACEHOLDER-->';
 
+	/**
+	 * Template types.
+	 *
+	 * @var array
+	 */
+	protected static $template_types = array(
+		'paged', // Deprecated.
+		'index',
+		'404',
+		'archive',
+		'author',
+		'category',
+		'tag',
+		'taxonomy',
+		'date',
+		'home',
+		'front_page',
+		'page',
+		'search',
+		'single',
+		'embed',
+		'singular',
+		'attachment',
+	);
+
+	/**
+	 * Initialize.
+	 */
+	public static function init() {
+		require_once AMP__DIR__ . '/includes/amp-post-template-actions.php';
+		if ( amp_is_canonical() ) {
+			$is_amp_endpoint = ( false !== get_query_var( AMP_QUERY_VAR, false ) ); // Because is_amp_endpoint() now returns true if amp_is_canonical().
```

Comment:
There isn't really a need to store this in a variable, the condition could be added directly in the if statement below.